UefiTestingPkg/FunctionalSystemTests/MemmapAndMatTestApp/MemmapAndMatTestApp.c

```diff
@@ -172,49 +172,41 @@ NoMatMapEntriesShouldHaveZeroSize (
 
 UNIT_TEST_STATUS
 EFIAPI
-AllLegacyMapEntriesShouldBePageAligned (
+AllEfiMemoryMapEntriesShouldBeAligned (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
   UNIT_TEST_STATUS       Status = UNIT_TEST_PASSED;
   UINTN                  Index;
   EFI_MEMORY_DESCRIPTOR  *Descriptor;
 
-  for (Index = 0; Index < mLegacyMapMeta.EntryCount; Index++) {
-    Descriptor = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)mLegacyMapMeta.Map + (Index * mLegacyMapMeta.EntrySize));
+  for (Index = 0; Index < mEfiMemoryMapMeta.EntryCount; Index++) {
+    Descriptor = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)mEfiMemoryMapMeta.Map + (Index * mEfiMemoryMapMeta.EntrySize));
     if (((Descriptor->PhysicalStart & EFI_PAGE_MASK) != 0) ||
         ((Descriptor->VirtualStart & EFI_PAGE_MASK) != 0))
     {
       Status = UNIT_TEST_ERROR_TEST_FAILED;
       break;
     }
-  }
-
-  return Status;
-} // AllLegacyMapEntriesShouldBePageAligned()
-
-UNIT_TEST_STATUS
-EFIAPI
-AllMatMapEntriesShouldBePageAligned (
-  IN UNIT_TEST_CONTEXT  Context
-  )
-{
-  UNIT_TEST_STATUS       Status = UNIT_TEST_PASSED;
-  UINTN                  Index;
-  EFI_MEMORY_DESCRIPTOR  *Descriptor;
 
-  for (Index = 0; Index < mMatMapMeta.EntryCount; Index++) {
-    Descriptor = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)mMatMapMeta.Map + (Index * mMatMapMeta.EntrySize));
-    if (((Descriptor->PhysicalStart & EFI_PAGE_MASK) != 0) ||
-        ((Descriptor->VirtualStart & EFI_PAGE_MASK) != 0))
+    // per the UEFI spec, these types need to have
+    // RUNTIME_PAGE_ALLOCATION_GRANULARITY
+    if ((Descriptor->Type == EfiRuntimeServicesCode) ||
+        (Descriptor->Type == EfiRuntimeServicesData) ||
+        (Descriptor->Type == EfiACPIMemoryNVS) ||
+        (Descriptor->Type == EfiReservedMemoryType))
     {
-      Status = UNIT_TEST_ERROR_TEST_FAILED;
-      break;
+      if (((Descriptor->PhysicalStart & (RUNTIME_PAGE_ALLOCATION_GRANULARITY - 1)) != 0) ||

```

Comment:
I have some concerns about this implementation.

now that RUNTIME_PAGE_ALLOCATION_GRANULARITY define can be changed, including this in the code directly means that this has to be compiled on the system that it is targeting, or on a system that is similar enough in settings.

Since RUNTIME_PAGE_ALLOCATION_GRANULARITY can not be modified, I think we should move it to either an input parameter for the utility, or it should be determined through some runtime mechanism.

Reply:
Hmm, I actually think that it is good that we target RUNTIME_PAGE_ALLOCATION_GRANULARITY here. That is UEFI spec defined. So while there are some older systems that cannot support say 64k as a RPAG, they are breaking the UEFI spec. I don't think this test should pass for them. Although I guess this would lead to inconsistent results, if you compiled on a standard system, the test would break, if you compiled against the old system, it would pass. I still sort of thing that's okay. If you run your own test, fine it works. If you run the standard compiled test, you see you are breaking the spec. The other piece here is that all new platforms should be using the standard RPAG, so this should not be an issue going forward. So I would be inclined to leave this as is, from the test perspective, directly tested against RPAG is the right thing to do.